pkg/component/extensions/operatingsystemconfig/original/original.go

```diff
@@ -30,28 +30,17 @@ import (
 	"github.com/gardener/gardener/pkg/features"
 )
 
-// ComponentsFn is a function that returns the list of original operating system config components.
 // Exposed for testing.
-var ComponentsFn = Components
+var (
+	// ComponentsFn is a function that returns the list of original operating system config components.
+	ComponentsFn = Components
+	// ProvisionComponentsFn is a function that returns the list of original operating system config components for `provision` purpose.
+	ProvisionComponentsFn = ProvisionComponents
```

Comment:
TBH, I have some difficulties with this 😅 (I just understand it because I know the code, but I can imagine that this is confusing as hell for new developers (especially in the light that this whole OSC generation already has its own complexity)).

The `original` package was always meant to solely contain components that get part of the "original OSC", i.e. the one with `purpose=reconcile`. With this introduction, things are getting mixed up. Also, it encourages to further extend the list of "provision components" (which is actually something we don't want in almost all cases (this PR is the only/first exception since introduction of Gardener)).

I would prefer to keep things simple and just call `gardeneruser.New().Config()` and `sshdensurer.New().Config` [here](https://github.com/gardener/gardener/pull/9077/files#diff-ac1ca99561871a6343b6c889961a1ec9ceea95aa06a88a0a44ccd395f12c19d7R694) and add their units and files to the OSC.

Reply:
Next try 😉